model/src/lib.rs

```diff
@@ -52,26 +58,97 @@ pub struct OrderCreation {
 }
 
 impl OrderCreation {
+    pub const ORDER_TYPE_HASH: [u8; 32] =
+        hex!("b71968fcf5e55b9c3370f2809d4078a4695be79dfa43e5aa1f2baa0a9b84f186");
+
     pub fn token_pair(&self) -> Option<TokenPair> {
         TokenPair::new(self.buy_token, self.sell_token)
     }
-    pub fn order_owner(&self) -> H160 {
-        // dummy implementation, waiting for Valentins PR
-        H160::zero()
+
+    pub fn order_digest(&self) -> [u8; 32] {
+        let mut hash_data = [0u8; 320];
+        hash_data[0..32].copy_from_slice(&Self::ORDER_TYPE_HASH);
```

Comment:
Made an issue for this because I don't want to bog down this PR with it. When I implemented this I found it easier to set the bytes directly but agree we can see if there is an easy way to reuse those crates.